### PR TITLE
Update issue templates, commitcheck and Contributing.md

### PR DESCRIPTION
**Pull request by Ornias1993**
### Motivation and Context
This PR aims to seperate the non-controversial changes from #10853 and squashes those together for a quick-and-easy merge.
It removes unused content from the commit check and contributing.md
and
also adds links to the mailinglists/irc to the issue creator.

Thats all.

### Description
- Removes OpenZFS ports from commit check
- Removes OpenZFS ports from CONTRIBUTING.md
- Adds mailingslists and irc to issue template selector
- Remove blank issue option from issue creator

### How Has This Been Tested?
Manually, seperate repo, like #10853

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [X] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation (a change to man pages or other documentation)

### Checklist:

- [X] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [X] I have updated the documentation accordingly.
- [X] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [X] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [X] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

Signed-off-by: Kjeld Schouten-Lebbing <kjeld@schouten-lebbing.nl>
Requires-builders: style

### CODE REVIEW

**Comment by Ornias1993** [comment]
@behlendorf I think this can be easier merged than my previous proposal.
I'll update the other proposal as soon as we get this merged and will make sure to also send that other one into the mailinglist for design review afterwards.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+blank_issues_enabled: false
+contact_links:
+  - name: OpenZFS Community Support Mailinglist (linux)
+    url: https://zfsonlinux.topicbox.com/groups/zfs-discuss
+    about: Get community support for OpenZFS on Linux
+  - name: FreeBSD Community Support Mailinglist
+    url: https://lists.freebsd.org/mailman/listinfo/freebsd-fs
+    about: Get community support for OpenZFS on FreeBSD
+  - name: OpenZFS on IRQ
```
Comment (gmelikov) [inline .github/ISSUE_TEMPLATE/config.yml:9]:
```suggestion
  - name: OpenZFS on IRC
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    url: https://webchat.freenode.net/#openzfs
+    about: Use IRQ to get community support for OpenZFS
```
Comment (gmelikov) [inline .github/ISSUE_TEMPLATE/config.yml:11]:
```suggestion
    about: Use IRC to get community support for OpenZFS
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+blank_issues_enabled: false
+contact_links:
+  - name: OpenZFS Community Support Mailinglist (linux)
```
Comment (behlendorf) [inline .github/ISSUE_TEMPLATE/config.yml:3]: "Mailinglist" -> "Mailing List"
linux -> Linux

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    url: https://zfsonlinux.topicbox.com/groups/zfs-discuss
+    about: Get community support for OpenZFS on Linux
+  - name: FreeBSD Community Support Mailinglist
```
Comment (behlendorf) [inline .github/ISSUE_TEMPLATE/config.yml:6]: "Mailinglist" -> "Mailing List"

**Review by behlendorf — COMMENTED** [review]
Thanks for splitting this off.

**Review by behlendorf — APPROVED** [review]

**Review by gmelikov — APPROVED** [review]

**Pull request merged**